mlx/coverity_services.py

```diff
@@ -1,614 +1,396 @@
-                                         CLASSIFICATION_LIST,
-                                         filter_spec.classificationNameList)
+        Args:
+            url (str): The URL for the request
+            data (dict): Optional data to send
 
-        # apply any filter on action
-        if filters['action']:
-            self.handle_attribute_filter(filters['action'], 'Action', ACTION_LIST, filter_spec.actionNameList)
+        Returns:
+            dict: the content of server's response
 
-        # apply any filter on Components
-        if filters['component']:
-            self.handle_component_filter(filters['component'], filter_spec)
+        Raises:
+            requests.HTTPError
+        """
+        if data:
+            response = self.session.post(url, json=data)
+        else:
+            response = self.session.get(url)
+        if response.ok:
+            return response.json()
+        try:
+            err_msg = response.json()["message"]
+        except (requests.exceptions.JSONDecodeError, KeyError):
+            err_msg = response.content.decode()
+        self.logger.warning(err_msg)
+        return response.raise_for_status()
 
-        # apply any filter on CWE values
-        if filters['cwe']:
-            self.handle_attribute_filter(filters['cwe'], 'CWE', None, filter_spec.cweList)
+    @staticmethod
+    def validate_filter_option(name, req_csv, valid_attributes, allow_regex=False):
+        """Add filter when the attribute is valid. If `valid_attributes` is empty or falsy,
+        all attributes of the CSV list are valid.
+        The CSV list can allow regular expressions when `allow_regex` is set to True.
 
-        # apply any filter on CID values
-        if filters['cid']:
-            self.handle_attribute_filter(filters['cid'], 'CID', None, filter_spec.cidList)
+        Args:
+            name (str): String representation of the attribute.
+            req_csv (str): A CSV list of attribute values to query.
+            valid_attributes (list/dict): The valid attributes.
+            allow_regex (bool): True to treat filter values as regular expressions, False to require exact matches
 
-        # if a special custom attribute value requirement
-        if custom:
-            self.handle_custom_filter_attribute(custom, filter_spec)
+        Returns:
+            list[str]: The list of valid attributes
+        """
+        logging.info("Validate required %s [%s]", name, req_csv)
+        filter_values = []
+        for field in req_csv.split(","):
+            if not valid_attributes or field in valid_attributes:
+                logging.info("Classification [%s] is valid", field)
+                filter_values.append(field)
+            elif allow_regex:
+                pattern = re.compile(field)
+                for element in valid_attributes:
+                    if pattern.search(element) and element not in filter_values:
+                        filter_values.append(element)
```

Comment:
a `set` would come in handy here: no need to ensure uniqueness via `and element not in filter_values` then

Reply:
Resolved in b48d82a16245b83b60feafddaecca8c0ad0d191e